app/Http/Controllers/ContestsController.php

```diff
@@ -134,4 +138,41 @@ public function export(Contest $contest)
         $csv = $contest->getCSVExport();
         $csv->output('contest' . $contest->id . '.csv');
     }
+
+    /**
+     * Grab the form to add a user retroactively.
+     *
+     * @param  \Gladiator\Models\Contest $contest
+     * @return  \Illuminate\Http\Response
+     */
+    public function signupForm(Contest $contest)
+    {
+        $contest = $contest->load(['waitingroom', 'competitions']);
+
+        return view('contests.signup', compact('contest'));
+    }
+
+    /**
+     * The post request to add a user retroactively.
+     *
+     * @param  SignupUserRequest $request
+     * @return \Illuminate\Http\Response
+     */
+    public function signupUser(SignupUserRequest $request)
+    {
+        $user = $this->registrar->findUserAccount($request->all());
+
+        if (! $user instanceof User) {
+            $user = $this->registrar->createUser($user);
+        }
+
+        // Add the user to a competition or waiting room, but only if not already in it.
+        if ($request->has('competition_id') && ! $user->competitions()->where('id', $request->competition_id)->first()) {
```

Comment:
I'll move it to a manager, but it doesn't fail. if it finds the user it just silently ignores  the request and shows the success message, since the person was already in the waiting room/comp